### ISSUE TITLE
Elemental2 wrapper for DOMParser

### ISSUE THREAD

**Issue by jiakuan** [google/elemental2#157]
Can't find `DOMParser` from the current elemental2 library.
https://developer.mozilla.org/en-US/docs/Web/API/DOMParser

Shall we support it as well?

**Comment by gkdn** [comment]
@jDramaix This is defined in https://github.com/google/closure-compiler/blob/master/externs/browser/gecko_xml.js though I don't see us using any extern named gecko (though this one doesn't seem Mozilla specific)

**Comment by niloc132** [comment]
When I asked the closure-compiler maintainers some years ago about the different browser-named extern files, they indicated that this was often the original source of that type, and that it wouldn't be beneficial to migrate the externs out of that file and into something common now that the type is standardized.

**Comment by jDramaix** [comment]
Yes we do not use any specific browser extern files. If the api has been standardized we need to move the api in another extern file (w3c_xml.js in this case)

**Comment by niloc132** [comment]
It looks like my memory is faulty or policy has changed: I can't find the specific conversation where gwt community members were asked to not bulk migrate standardized entries, but https://github.com/google/closure-compiler/issues/2478 is one such conversation. 

The entire gecko_dom.js is standardized now (with the exception of the serializer method `serializeToStream`, which is no longer supported by any browser that I can see), but as it is only a few members, so I'll see about moving it.

**Comment by jDramaix** [comment]
The problem is that `DOMParser.parseFromString()` is accepting a  [`TrustedHTML`](https://developer.mozilla.org/en-US/docs/Web/API/TrustedHTML) and this is not supported in Elemental2 yet.

**Comment by zbynek** [comment]
@jDramaix elemental also includes w3c_range that uses TrustedHTML, is that also a problem? https://github.com/google/closure-compiler/blob/46131b1844fb070316a98c5b4c047ee3d786553e/externs/browser/w3c_range.js#L250
https://github.com/google/elemental2/blob/33cea8e04854d3b6839c7ff394e865567a574832/java/elemental2/dom/BUILD#L66

**Comment by jDramaix** [comment]
The definition of the types are there but there is no J2CL/GWT way to instantiate them (I don't even know a standard way to instantiate them on JS). Also, the version of `DOMParser.parseFromString()` is considered as unsafe and the Elemental2 build will fail internally at Google. So we would end up generating Java code only targeting the signature using TrustedHTML. It' not a working solution as it would not be usable by most users.

**Comment by zbynek** [comment]
> I don't even know a standard way to instantiate them on JS)

So is this issue blocked by FireFox+Safari support for trusted types? (https://github.com/mozilla/standards-positions/issues/20, https://webkit.org/standards-positions/ doesn't even mention it).
Elemental already contains the code needed for instantiating trusted HTML in Chromium

```
TrustedTypePolicyOptions tpo = TrustedTypePolicyOptions.create();
tpo.setCreateHTML((str, ignore) -> str);
DomGlobal.trustedTypes.createPolicy("unsafe", tpo).createHTML("<img>");
```

**Comment by zbynek** [comment]
2025 update: preview implementations for Trusted Types now exist in both Safari and FireFox
https://developer.mozilla.org/en-US/docs/Web/API/Trusted_Types_API
